youtube_dl/extractor/common.py

```diff
@@ -1139,7 +1139,8 @@ def _extract_m3u8_formats(self, m3u8_url, video_id, ext=None,
                 if m3u8_id:
                     format_id.append(m3u8_id)
                 last_media_name = last_media.get('NAME') if last_media and last_media.get('TYPE') != 'SUBTITLES' else None
-                format_id.append(last_media_name if last_media_name else '%d' % (tbr if tbr else len(formats)))
+                if last_media_name or not live:
```

Comment:
>Should be and

We may result in identical format ids in that case. Though there might be several streams without `NAME` too. What about simple counter? E.g. `fid`, `fid-1`, `fid-2`.